apps/web/app/input-cluster.tsx

```diff
@@ -1,21 +1,25 @@
 'use client'
 
-import { processInputFile } from 'cli'
+import { type Cluster, processInputFile } from 'cli'
 import React, { useEffect, useState } from 'react'
-import type { Cluster } from 'cli'
 import { ClusterComponent } from './cluster'
-import { assignColorIndex, isMinimumLeafsReached } from './assign-color'
+import {
+  assignColorIndex,
+  getNumberOfLeafsInCluster,
+  isMinimumLeafsReached,
+} from './assign-color'
 import { LeafsQuantitySlider } from './quantity-leaf-slider'
 
 export function InputCluster(): JSX.Element {
-  const [processedCluster, setProcessedCluster] = useState<
-    Cluster | undefined
-  >()
-  const [maxLeafsQuantity, setMaxLeafsQuantity] = useState(15)
-  const [minLeafsQuantity, setMinLeafsQuantity] = useState(2)
+  const [csvText, setCSVText] = useState<string | null>()
+  const [processedCluster, setProcessedCluster] = useState<Cluster>()
+  const [maxLeafsPerCluster, setMaxLeafsPerCluster] = useState<number>()
+  const [minLeafsPerCluster, setMinLeafsPerCluster] = useState(2)
   const [colorMutation, setColorMutation] = useState(0)
   const [absoluteCorrelation, setAbsoluteCorrelation] = useState(false)
-  const [csvText, setCSVText] = useState<string | null>()
+
+  const maxNumberOfLeafs =
+    processedCluster && getNumberOfLeafsInCluster(processedCluster)
```

Comment:
```suggestion
    processedCluster && getNumberOfLeafsInCluster(processedCluster)
```

&& is not clear here.